### PR TITLE
Docker command changes

### PR DESCRIPTION
**Pull request by walkero-gr**
- Changes on the docker command, so that it uses the phpunit installed in a docker and not always use the phpunit under vendor folder.
- Added xmlConfigFileName parameter, so that the user can set a totally different one. If this is not set defaults to phpunit.xml

### CODE REVIEW

**Comment by calebporzio** [comment]
I don't use Docker so I can't tell if this is warranted or not. @sixlive ?

**Comment by sixlive** [comment]
## Docker file command
My initial gut reaction is that this is pretty edge case and very uncommon. @walkero-gr, can you provide some more context around the problem this solves? Where else would the PHPUnit be where its not in the vendor directory?

## Config File name
I also feel that this is rather uncommon but I think this makes sense and is reasonable to merged.

**Comment by calebporzio** [comment]
Thanks for weighing in TJ!

**Comment by walkero-gr** [comment]
Thank you for taking the time to check my pull request.

@sixlive Docker gives the ability to have single logical services per container, so phpunit might be on it's own container, so that could be easier to experiment with different versions of it. If you use only the one that exists under vendor folder, this cannot be changed easily, unless you change the composer installation script.

Maybe a good way to cover both cases would be to set an optional configuration parameter, that can override the default one.

**Comment by sixlive** [comment]
> so that could be easier to experiment with different versions of it

I think that makes sense. Thanks for the clarification!

File: src/docker-phpunit-command.js [hunk 1]
```diff
@@ -6,6 +6,27 @@ module.exports = class DockerPhpUnitCommand extends RemotePhpUnitCommand {
...
+        }
+    }
+
+    get xmlConfigFileName() {
+        if (this.config.get("xmlConfigFileName")) {
+            return this.config.get("xmlConfigFileName");
+        } 
+
+        return 'phpunit.xml';
+    }
```
Comment (calebporzio) [inline src/docker-phpunit-command.js]: Seems odd to include this config option and only use if for docker runs. As an end user not using docker I would assume it would work for everything.

Thinking this shouldn't be part of this PR - adding a separate config file option is a separate issue. (one that frankly I'm more open too than the remote phpunit binary)

File: src/docker-phpunit-command.js [hunk 2]
```diff
@@ -18,6 +39,13 @@ module.exports = class DockerPhpUnitCommand extends RemotePhpUnitCommand {
...
     wrapCommand(command) {
-        return `${this.dockerCommand} ${command}`;
+        var runCommand = `${this.dockerCommand}`;
+        runCommand += `${this.configuration}`;
+        runCommand += ' ' + `${this.file}`;
+        if (this.filter) {
+            runCommand += ' ' + `${this.filter}`;
+        }
+
+        return runCommand;
```
Comment (calebporzio) [inline src/docker-phpunit-command.js]: This block also makes me nervous because we are no longer actually "wrapping" the command - you are re-constructing it, so changes that are made to the base command generator will not trickle down and impact docker users.

**Comment by calebporzio** [comment]
After looking through the code, unfortunately, I think I'm going to close this for now. Is it possible to achieve what you want by hacking existing config options? 

Also, I'm open to a phpunit.xml config option in another PR if it works for all runs (not just docker)

Thanks for being willing to contribute the plugin!
